### DISCUSSION TITLE
Reference a fixed JSON file

### DISCUSSION THREAD

**Discussion by VaniSachdev** [microsoft/vscode-discussions#42] [Extension Development QnA] [answered]
I’m following along the [tree-view sample](https://github.com/microsoft/vscode-extension-samples/blob/main/tree-view-sample/src/jsonOutline.ts), and my goal is for the JSON file the tree view is created from to be hard coded into the extension (the path of a specific JSON file) such that regardless of which file is open the tree view is the same. Currently the code is dependent on the active editor being a JSON file, and I assumed it would be an easy proof of concept if I could hardcode the active editor to reference a specific path somewhere in the constructor but I haven't been able to figure it out. I'm fairly new to developing VS Code extensions (& typescript!), so any help would be great!

**Comment by alefragnani** [comment, marked as answer]
I suppose you are looking for customizing the `parseTree` function, which in that sample, is responsible to _load the JSON file from the active editor_, as you can see:

```
	private parseTree(): void {
		this.text = '';
		this.tree = null;
		this.editor = vscode.window.activeTextEditor;
		if (this.editor && this.editor.document) {
			this.text = this.editor.document.getText();  <<-- HERE
			this.tree = json.parseTree(this.text);
		}
	}
```

https://github.com/microsoft/vscode-extension-samples/blob/128179bd61c0e0a24322ea42d2003b6ace5bc94e/tree-view-sample/src/jsonOutline.ts#L81-L89

In your case, line 86 it the one responsible for that. Replace it with your custom JSON file and you are good to go.

Hope this helps
